client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-how-to-migrate/index.tsx

```diff
@@ -1,8 +1,9 @@
 import { StepContainer } from '@automattic/onboarding';
 import { useTranslate } from 'i18n-calypso';
-import { FC, useMemo } from 'react';
+import { FC, useMemo, useState } from 'react';
```

Comment:
The component `SiteMigrationHowToMigrate` is getting a little complex (big). I started extracting part of the logic to another hook [here](https://github.com/Automattic/wp-calypso/pull/95413/files#diff-d840edfeb1289157ef7b061afe8ba90034b8a354210586b68aacac10c2d60f50). I think it would be nice to base this PR there and also add the loading logic there, returning the loading state. WDYT?

Reply:
And I thought of another scenario where we could have the same issue.
If the user is in a slow connection and they click on one of the options quickly it would be registering the "pending" sticker, and could try to register the "pending-{TYPE}" at the same time.

If we rebase this PR, we could also add the loader for this scenario: https://github.com/Automattic/wp-calypso/pull/95413/files#diff-d840edfeb1289157ef7b061afe8ba90034b8a354210586b68aacac10c2d60f50R26

If we follow the suggestion of the other comment, I think it would assume this behavior automatically.

Reply:
Sounds good!

I've updated the code but I'm a bit unsure if my implementation is aligned with your vision. 😅 Could you give it a look again?